pkg/controllers/reconcilers.go

```diff
@@ -54,14 +54,34 @@ type Config struct {
 	Tracker   tracker.Tracker
 }
 
+// ObjectType represents the type of a Kubernetes object. Fields of type ObjectType are
+// initialized with an empty object and not modified subsequently.
+type ObjectType struct {
+	runtime.Object
+}
+
+// NewObjectType constructs an ObjectType from the given empty object.
+func NewObjectType(obj runtime.Object) ObjectType {
+	return ObjectType{obj}
+}
+
+// asObject returns a deep copy of the object type as an apis.Object.
+func (ot ObjectType) asObject() apis.Object {
```

Comment:
should be exported, or drop the method and lean on the current `Object` field

```suggestion
func (ot ObjectType) Object() apis.Object {
```